datashader/core.py

```diff
@@ -514,7 +514,7 @@ def bypixel(source, canvas, glyph, agg):
         cols_to_keep[glyph.x] = True
         cols_to_keep[glyph.y] = True
         if hasattr(glyph, 'z'):
-            cols_to_keep[glyph.z] = True
+            cols_to_keep[glyph.z[0]] = True
```

Comment:
Where did this come from?  Seems odd that it shows up in this PR's tests?

Reply:
Previously this code was called conditionally, i.e. only for dataframes with categorical columns and the tests weren't testing that.

Reply:
Ok, thanks.